src/db.test.js

```diff
@@ -0,0 +1,42 @@
+import { store } from './store'
+import * as db from './db'
+import { initialize } from './initialize'
+import { hashThought } from './util'
+
+beforeAll(async () => {
+  await initialize()
+})
+
+it('load settings into indexedDB on initialization', async () => {
+  const hash = hashThought('Settings')
+
+  const thoughtState = store.getState().thoughts.thoughtIndex[hash]
+
+  expect(thoughtState).not.toBeUndefined()
+  expect(thoughtState.contexts.length).toEqual(1)
+
+  const thoughtDB = await db.getThought(hash)
+
+  expect(thoughtDB).not.toBeUndefined()
+  expect(thoughtDB.contexts.length).toEqual(1)
+
+  expect(thoughtState.contexts[0].id).toEqual(thoughtDB.contexts[0].id)
+})
+
+it('presist data on indexedDB after adding new thought', async () => {
+
+  await store.dispatch({ type: 'newThought', value: 'db-test' })
+  const hash = hashThought('db-test')
+
+  const thoughtState = store.getState().thoughts.thoughtIndex[hash]
+
+  expect(thoughtState).not.toBeUndefined()
+  expect(thoughtState.contexts.length).toEqual(1)
+
+  const thoughtDB = await db.getThought(hash)
+
+  expect(thoughtDB).not.toBeUndefined()
+  expect(thoughtDB.contexts.length).toEqual(1)
+
+  expect(thoughtState.contexts[0].id).toEqual(thoughtDB.contexts[0].id)
+})
```

Comment:
This test appears to be identical to the first test.

Reply:
The first one checks if the Settings gets synced in the db on load. The second one checks if a new thought added gets synced to the db. They are kinda identical. I will remove the later one.